src/components/Story.svelte

```diff
@@ -1,39 +1,69 @@
-<script>
-  import { getStoryRenderContext, useContext } from './context';
+<script lang="ts" generics="M extends Meta">
+  import type { Meta, StoryObj, StoryContext } from '@storybook/svelte';
+  import type { Snippet } from 'svelte';
 
-  const context = useContext();
+  import { useStoriesExtractor, useStoryRenderer, useStoriesTemplate } from './context.svelte.js';
 
-  export let name;
-  export let template = null;
-  export let play = null;
+  type Props = {
+    children?: Snippet<[StoryObj<M>["args"], StoryContext<M["args"]>]>;
+    /**
+    * Id of the story.
+    *
+    * Optional, auto-generated from name if not specified.
+    */
+    id?: string;
+    /**
+    * Name of the story.
+    * @default "Default"
+    */
+    name?: string;
+    /**
+     * @deprecrated
+     * Use `tags={['autodocs']}` instead.
+     */
+    autodocs?: never;
+    // TODO: Discuss if this is still neeeded
+    /**
+     *
+     * Specify which sources should be shown.
+     *
+     * By default, sources for an args story are auto-generated.
+     * If source is true, then the source of the story will be used instead.
+     * If source is a string, it replaces the source of the story.
+     */
+    source?: boolean | string;
+  } & StoryObj<M>;
 
-  if (!name) {
-    throw new Error('Missing Story name');
-  }
+  const { children, name = "Default", id, play, ...restProps }: Props = $props();
 
-  context.register({
-    name,
-    ...$$restProps,
-    play,
-    template: template != null ? template : !$$slots.default ? 'default' : null,
-  });
+  const extractor = useStoriesExtractor<M>();
+  const renderer = useStoryRenderer<M>();
+  const template = useStoriesTemplate<M>();
 
-  $: render = context.render && !context.templateName && context.storyName == name;
-  const ctx = getStoryRenderContext();
-  const args = ctx.argsStore;
-  const storyContext = ctx.storyContextStore;
+  const isCurrentlyViewed = $derived(!extractor.isExtracting && renderer.currentStoryName === name);
 
-  function injectIntoPlayFunction(ctxt, play) {
-    if (play && ctxt.playFunction) {
-      ctxt.playFunction.__play = play;
-    }
+  if (extractor.isExtracting) {
+    // @ts-expect-error FIXME: No idea how to satisfy this one, I could've missed something
```

Comment:
Potential bug, or not(?). TypeScript not being happy. I couldn't figure this one.